### PR TITLE
Extend fs' `getFileShasum`

### PR DESCRIPTION
**Pull request by Mitko-Kerezov**
So that it can use a custom-set algorithm and encoding
In addition, add some interfaces, describing what Apple's Content Delivery API returns
Ping @rosen-vladimirov

### CODE REVIEW

File: declarations.d.ts [hunk 1]
```diff
@@ -4,10 +4,120 @@ interface Object {
 
 interface IStringDictionary extends IDictionary<string> { }
 
+
+/**
+ * Describes iTunes Connect application types
+ */
+interface IItunesConnectApplicationType {
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: maybe `IiTunesConnectApplicationType` like `IiOSDevice`?

File: declarations.d.ts [hunk 2]
```diff
@@ -4,10 +4,120 @@ interface Object {
...
+	 * Applications developed for Mac OS
+	 * @type {string}
+	 */
+	Mac: string;
+}
+
+/**
+ * Descibes iTunes Connect applications
+ */
+interface IItunesConnectApplication {
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: same as above

File: declarations.d.ts [hunk 3]
```diff
@@ -4,10 +4,120 @@ interface Object {
...
+	type: string;
+	/**
+	 * Application's current version
+	 * @type {string}
+	 */
+	version: string;
+}
+
+/**
+ * Describes data returned from querying itunes' contentdelivery api
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: `contentdelivery`?

File: declarations.d.ts [hunk 4]
```diff
@@ -4,10 +4,120 @@ interface Object {
...
+ */
+interface IContentDeliveryBody {
+	/**
+	 * Error object - likely present if result's Success is false.
+	 */
+	error?: {
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: `error?: Error` maybe

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by rosen-vladimirov** [comment]
:+1:  after fixing comments

File: declarations.d.ts [hunk 5]
```diff
@@ -4,10 +4,109 @@ interface Object {
...
+		/**
+		 * Error message - likely present if Success is false.
+		 * @type {string}
+		 */
+		ErrorMessage?: string;
+		/**
+		 * Error message - likely present if Success is false.
+		 * @type {string[]}
+		 */
+		Errors?: string[];
```
Comment (rosen-vladimirov) [inline declarations.d.ts]: `string[]` or `Error[]`

Comment (Mitko-Kerezov) [inline declarations.d.ts]: `string[]`, as far as I can tell - the errors I got while testing were strings inside an array

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**